crates/sui-types/src/storage.rs

```diff
@@ -969,3 +969,138 @@ impl Display for DeleteKind {
+        self.0.get_lowest_available_checkpoint()
+    }
+
+    fn get_full_checkpoint_contents_by_sequence_number(
+        &self,
+        sequence_number: CheckpointSequenceNumber,
+    ) -> Result<Option<FullCheckpointContents>, Self::Error> {
+        self.0
+            .get_full_checkpoint_contents_by_sequence_number(sequence_number)
+    }
+
+    fn get_full_checkpoint_contents(
+        &self,
+        digest: &CheckpointContentsDigest,
+    ) -> Result<Option<FullCheckpointContents>, Self::Error> {
+        self.0.get_full_checkpoint_contents(digest)
+    }
+
+    fn get_committee(&self, epoch: EpochId) -> Result<Option<Arc<Committee>>, Self::Error> {
+        self.0.get_committee(epoch)
+    }
+
+    fn get_transaction_block(
+        &self,
+        digest: &TransactionDigest,
+    ) -> Result<Option<VerifiedTransaction>, Self::Error> {
+        self.0.get_transaction_block(digest)
+    }
+
+    fn get_transaction_effects(
+        &self,
+        digest: &TransactionEffectsDigest,
+    ) -> Result<Option<TransactionEffects>, Self::Error> {
+        self.0.get_transaction_effects(digest)
+    }
+
+    fn get_transaction_events(
+        &self,
+        digest: &TransactionEventsDigest,
+    ) -> Result<Option<TransactionEvents>, Self::Error> {
+        self.0.get_transaction_events(digest)
+    }
+}
+
+impl WriteStore for OneShotSharedInMemoryStore {
+    fn insert_checkpoint(&self, checkpoint: &VerifiedCheckpoint) -> Result<(), Self::Error> {
+        {
+            let mut locked = self.0 .0.write().unwrap();
+            locked.checkpoints.clear();
+            locked.sequence_number_to_digest.clear();
+        }
+        self.0.insert_checkpoint(checkpoint)?;
+        Ok(())
+    }
+
+    fn update_highest_synced_checkpoint(
+        &self,
+        checkpoint: &VerifiedCheckpoint,
+    ) -> Result<(), Self::Error> {
+        self.0.update_highest_synced_checkpoint(checkpoint)?;
+        Ok(())
+    }
+
+    fn update_highest_verified_checkpoint(
+        &self,
+        checkpoint: &VerifiedCheckpoint,
+    ) -> Result<(), Self::Error> {
+        self.0.update_highest_verified_checkpoint(checkpoint)?;
+        Ok(())
+    }
+
+    fn insert_checkpoint_contents(
+        &self,
+        checkpoint: &VerifiedCheckpoint,
+        contents: VerifiedCheckpointContents,
+    ) -> Result<(), Self::Error> {
+        {
+            let mut locked = self.0 .0.write().unwrap();
+            locked.transactions.clear();
+            locked.effects.clear();
+            locked.contents_digest_to_sequence_number.clear();
+            locked.full_checkpoint_contents.clear();
+            locked.checkpoint_contents.clear();
+        }
+        self.0.insert_checkpoint_contents(checkpoint, contents)?;
+        Ok(())
+    }
+
+    fn insert_committee(&self, new_committee: Committee) -> Result<(), Self::Error> {
+        self.0.insert_committee(new_committee)?;
```

Comment:
nit: just `self.0.insert_committee(new_committee)` instead of two lines method should work